polynote-kernel/src/main/scala/polynote/kernel/ScalaCompiler.scala

```diff
@@ -611,12 +609,15 @@ object ScalaCompiler {
     otherClasspath: List[File],
     modifySettings: Settings => Settings
   ): RIO[Config with System, ScalaCompiler] = for {
-    settings          <- ZIO(modifySettings(defaultSettings(new Settings(), dependencies.map(_.file) ++ otherClasspath)))
-    global            <- ZIO(new Global(settings, KernelReporter(settings)))
-    counter           <- kernelCounter
-    notebookPackage    = s"notebook$counter"
-    classLoader       <- makeClassLoader(settings, dependencies)
-    _                 <- ZIO(new SemanticdbPlugin(global))
+    config             <- Config.access
+    dependencyClasspath = dependencies.map(_.file)
+    compilerClasspath   = if (config.behavior.dependencyIsolation) dependencyClasspath ++ otherClasspath else otherClasspath ++ dependencyClasspath
```

Comment:
Huh, so is this actually fixing a bug in the dependency isolation implementation?